### ISSUE TITLE
In openIDConnect validator get_id_token is never called

### ISSUE THREAD

**Issue by dhrp** [idan/oauthlib#501]
First of; thanks for the recent addition of the OpenID Connect improvements! (@wiliamsouza)!

This is exactly what I need. 

However, I seem to have ran into a snag; the `get_id_token()` method of my own class never gets called. After careful tracing it appears that happens because there is something wrong with the order of execution.

In order words (code-snippets)

My own validator:
```
class OAuth2OpenIDConnectValidator(OAuth2Validator):
   def get_id_token(self, token, token_handler, request):
```

get_id_token() never gets called because

```
class OpenIDConnectAuthCode(OpenIDConnectBase):
    def __init__(self, request_validator=None, **kwargs):
       [ ... snip ... ]
       self.register_token_modifier(self.add_id_token)
```
happens *after*

```
def create_token_response(self, request, token_handler):
[ .. snip .. ]
        for modifier in self._token_modifiers:
            token = modifier(token, token_handler, request)
```

which is the place where the get_id_token() in my custom validator would be called. 

It's not 100% clear to me whether this is a bug, if it is because I use oauthlib through django-oauth-toolkit here; or that I'm doing something wrong.

Additionally; from the [documentation](https://oauthlib.readthedocs.io/en/stable/oauth2/oidc/validator.html) (note this is 'stable', not 'latest'?) I understood that there are four methods that need to be implemented; however, I noticed that I also had to override `validate_response_type` because otherwise the client auth would fail on an invalid response type.

**Comment by dhrp** [comment]
It looks like I was still doing things wrong and misunderstanding..

Some mistakes I made:
* the improvements by @wiliamsouza were merged, but the the issue with backwards incompatibilty was not fixed, but it was reverted in 2.06.
* I didn't properly understand that request type 'id_token' was (only) for the implicit grant types. 
* The implementation is not quite complete yet; I got parts to work, but only by disabling some checks

I'm now going to use django-oidc-provider instead of django-oauth-toolkit. It works out of the box and is more full-featured for oidc, even if the issues here were to be fixed.
